WebGPU] beginningOfPassWriteIndex and endOfPassWriteIndex default to 0 and 1, leading to CTS failures

mwyrzykowski opened the pull request:
#### 11388c6920cd0868d470f47e077349fa758256cd
<pre>
WebGPU] beginningOfPassWriteIndex and endOfPassWriteIndex default to 0 and 1, leading to CTS failures
<a href="https://bugs.webkit.org/show_bug.cgi?id=297808">https://bugs.webkit.org/show_bug.cgi?id=297808</a>
<a href="https://rdar.apple.com/158977287">rdar://158977287</a>

Reviewed by Tadeu Zagallo.

A QuerySet with 1 element is allowed, it will only write to either the start or end of the render
or compute pass. This was previously rejected in WebKit as we defaulted begin and end to 0 and 1.

Fixing this results in several more query timestamps tests passing.

* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/beginComputePass-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/beginRenderPass-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/copyTextureToTexture-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/copyTextureToTexture.spec.js:
(F.prototype.TestCopyTextureToTexture): Deleted.
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/index_access-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/dynamic_state-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/createRenderBundleEncoder-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/encoder_open_state-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/encoder_state-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/programmable/pipeline_bind_group_compat-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/queries/general-expected.txt:
Add updated expectations

* LayoutTests/platform/mac-wk2/TestExpectations:
Stop skipping tests which are now passing.

* Source/WebCore/Modules/WebGPU/GPUComputePassTimestampWrites.h:
* Source/WebCore/Modules/WebGPU/GPURenderPassTimestampWrites.h:
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUComputePassTimestampWrites.h:
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPURenderPassTimestampWrites.h:
* Source/WebKit/Shared/WebGPU/WebGPUComputePassTimestampWrites.h:
* Source/WebKit/Shared/WebGPU/WebGPURenderPassTimestampWrites.h:
Default to undefined query index instead of zero and one.

Canonical link: <a href="https://commits.webkit.org/299163@main">https://commits.webkit.org/299163@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ad657f30eab6f76ca39a3b936120ec40e26ea3e1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/118075 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37753 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28389 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124229 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/70113 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/5ae5fde8-90af-4df3-9c8c-c6e2e51490a1) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/119953 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38446 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46335 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89605 "Found 1 new test failure: fullscreen/fullscreen-zero-height-element-with-non-zero-children.html (failure)") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/59211 "") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/7baa1d7b-30b7-450c-864f-770be1d4f94b) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/121027 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30625 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/105875 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70098 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/33346e9d-d439-483f-a22b-73313ea9f75a) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/29693 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/23992 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/67894 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100050 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/24170 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127306 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/44978 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/33896 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98280 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45339 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102099 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98067 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24943 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43465 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21454 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/41423 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44850 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/50524 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44310 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/47655 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/45999 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/fa3759ee81a74c723d2168562955f347db37cc21. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49797

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d8c8f9cf83221faeafb7d3c4e7b466519992e279. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49797

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/ad657f30eab6f76ca39a3b936120ec40e26ea3e1. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49797

mwyrzykowski committed:
WebGPU] beginningOfPassWriteIndex and endOfPassWriteIndex default to 0 and 1, leading to CTS failures
https://bugs.webkit.org/show_bug.cgi?id=297808
rdar://158977287

Reviewed by Tadeu Zagallo.

A QuerySet with 1 element is allowed, it will only write to either the start or end of the render
or compute pass. This was previously rejected in WebKit as we defaulted begin and end to 0 and 1.

Fixing this results in several more query timestamps tests passing.

* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/beginComputePass-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/beginRenderPass-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/copyTextureToTexture-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/copyTextureToTexture.spec.js:
(F.prototype.TestCopyTextureToTexture): Deleted.
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/index_access-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/dynamic_state-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/createRenderBundleEncoder-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/encoder_open_state-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/encoder_state-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/programmable/pipeline_bind_group_compat-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/queries/general-expected.txt:
Add updated expectations

* LayoutTests/platform/mac-wk2/TestExpectations:
Stop skipping tests which are now passing.

* Source/WebCore/Modules/WebGPU/GPUComputePassTimestampWrites.h:
* Source/WebCore/Modules/WebGPU/GPURenderPassTimestampWrites.h:
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUComputePassTimestampWrites.h:
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPURenderPassTimestampWrites.h:
* Source/WebKit/Shared/WebGPU/WebGPUComputePassTimestampWrites.h:
* Source/WebKit/Shared/WebGPU/WebGPURenderPassTimestampWrites.h:
Default to undefined query index instead of zero and one.

Canonical link: https://commits.webkit.org/299163@main
